### PR TITLE
Remove FXIOS-5089 [v108] SnapKit from OneLineTableviewCell

### PR DESCRIPTION
**Pull request by yoanarios**
### [FXIOS-5089](https://mozilla-hub.atlassian.net/browse/FXIOS-5089)   #12145 

- Remove SnapKit from `OneLineTableviewCell`
- Call to applyTheme in `SiteTableViewHeader`
TODO: Create ticket to Set `leftImageView` and `titleLabel` to private and use proper configure on cell

### CODE REVIEW

File: Client/Frontend/Widgets/SiteTableViewHeader.swift [hunk 1]
```diff
@@ -67,6 +68,8 @@ class SiteTableViewHeader: UITableViewHeaderFooterView, ThemeApplicable, Reusabl
 
         showImage(model.isCollapsible)
         collapsibleState = model.collapsibleState
+
+        applyTheme(theme: theme)
```
Comment (lmarceau) [inline Client/Frontend/Widgets/SiteTableViewHeader.swift:72]: Theme is applied from the `applyTheme(theme:)` function already. This shouldn't be added, nor the theme = LightTheme.

Comment (lmarceau) [reply Client/Frontend/Widgets/SiteTableViewHeader.swift]: This should be fixed with https://github.com/mozilla-mobile/firefox-ios/pull/12249

File: Client/Frontend/Widgets/OneLineTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    private lazy var containerView: UIView = .build { _ in }
+
+    lazy var leftImageView: UIImageView = .build { imageView in
+        imageView.contentMode = .scaleAspectFit
+        imageView.layer.cornerRadius = 5.0
+        imageView.clipsToBounds = true
+    }
+
+    lazy var titleLabel: UILabel = .build { label in
+        label.textColor = .black
```
Comment (polymathruby) [inline Client/Frontend/Widgets/OneLineTableViewCell.swift:40]: If we're doing this in `applyTheme`, do we need it here?

File: Client/Frontend/Widgets/OneLineTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                                                 constant: -8),
+
+            bottomSeparatorView.leadingAnchor.constraint(equalTo: titleLabel.leadingAnchor),
+            bottomSeparatorView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            bottomSeparatorView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+            bottomSeparatorView.heightAnchor.constraint(equalToConstant: 0.7)
+        ])
+
+        selectedBackgroundView = selectedView
+        applyTheme()
```
Comment (polymathruby) [inline Client/Frontend/Widgets/OneLineTableViewCell.swift:114]: We're calling this here even though we're also calling it afterwards? Is there a reason?

File: Client/Frontend/Widgets/OneLineTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+    lazy var leftImageView: UIImageView = .build { imageView in
+        imageView.contentMode = .scaleAspectFit
+        imageView.layer.cornerRadius = 5.0
+        imageView.clipsToBounds = true
+    }
+
+    lazy var titleLabel: UILabel = .build { label in
+        label.textColor = .black
+        label.font = UIFont.systemFont(ofSize: 17, weight: .regular)
```
Comment (thatswinnie) [inline Client/Frontend/Widgets/OneLineTableViewCell.swift:41]: Should we use dynamic text here?

File: Client/Frontend/Widgets/OneLineTableViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                                            constant: 4),
+            titleLabel.bottomAnchor.constraint(equalTo: containerView.bottomAnchor,
+                                               constant: -4),
+            titleLabel.trailingAnchor.constraint(equalTo: containerView.trailingAnchor,
+                                                 constant: -8),
+
+            bottomSeparatorView.leadingAnchor.constraint(equalTo: titleLabel.leadingAnchor),
+            bottomSeparatorView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            bottomSeparatorView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+            bottomSeparatorView.heightAnchor.constraint(equalToConstant: 0.7)
```
Comment (thatswinnie) [inline Client/Frontend/Widgets/OneLineTableViewCell.swift:110]: This value should also be added to the UX enum.

File: Client/Frontend/Widgets/OneLineTableViewCell.swift (hunk 3 above)
Comment (yoanarios) [reply Client/Frontend/Widgets/OneLineTableViewCell.swift:114]: no I just didn't see in the clean up

File: Client/Frontend/Browser/SearchViewController.swift [hunk 6]
```diff
@@ -287,18 +287,23 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
...
+
         searchEngineScrollViewContent.addSubview(searchButton)
         searchButton.snp.makeConstraints { make in
-            make.size.equalTo(SearchViewControllerUX.FaviconSize)
+            make.width.height.equalTo(SearchViewControllerUX.FaviconSize)
             // offset the left edge to align with search results
-            make.leading.equalTo(leftEdge).offset(SearchViewControllerUX.SuggestionMargin * 2)
-            make.top.equalTo(self.searchEngineScrollViewContent).offset(SearchViewControllerUX.SuggestionMargin)
-            make.bottom.equalTo(self.searchEngineScrollViewContent).offset(-SearchViewControllerUX.SuggestionMargin)
+            make.leading.equalTo(leftEdge).offset(16)
```
Comment (lmarceau) [inline Client/Frontend/Browser/SearchViewController.swift:304]: nit: that 16 could be part of the SearchViewControllerUX?

File: Client/Frontend/Widgets/OneLineTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+        containerView.addSubviews(leftImageView,
+                                  titleLabel,
+                                  bottomSeparatorView)
+
+        contentView.addSubview(containerView)
+        bringSubviewToFront(containerView)
+
+        let containerViewTrailingAnchor = accessoryView?.leadingAnchor ?? contentView.trailingAnchor
+        let midViewLeadingMargin: CGFloat = shouldLeftAlignTitle ? 5 : 13
```
Comment (lmarceau) [inline Client/Frontend/Widgets/OneLineTableViewCell.swift:87]: nit, 5 and 13 could be part of UX

**Review by lmarceau — APPROVED** [review]

File: Client/Frontend/Browser/SearchViewController.swift (hunk 6 above)
Comment (yoanarios) [reply Client/Frontend/Browser/SearchViewController.swift:304]: I will fix in the ticket to remove SnapKit from SearchViewController

**Review by polymathruby — APPROVED** [review]

**Comment by yoanarios** [comment]
Screen Shot 2022-10-28 at 9 56 12 AM

**Pull request merged**